### PR TITLE
qa: Add stopatheight test

### PR DESCRIPTION
**Pull request by maflcko**
Motivated by  #10569

### CODE REVIEW

**Review by jonasschnelli — APPROVED** [review]
ACK fa75f235f14c05c7d7d433396f4f49809860340d

File: test/functional/blockchain.py [hunk 1]
```diff
@@ -34,14 +37,26 @@ def __init__(self):
...
         self._test_getchaintxstats()
         self._test_gettxoutsetinfo()
         self._test_getblockheader()
         self._test_getdifficulty()
         self._test_getnetworkhashps()
-        self.nodes[0].verifychain(4, 0)
+        self._test_stopatheight()
+        assert self.nodes[0].verifychain(4, 0)
+
+    def _test_stopatheight(self):
```
Comment (jnewbery) [inline test/functional/blockchain.py]: micronit: The other subtests are defined in the order that they're called. Perhaps place this definition after the other subtests?

**Review by jnewbery — COMMENTED** [review]
Tested ACK fa75f235f14c05c7d7d433396f4f49809860340d. Looks great!

Not sure that this entirely fits in blockchain.py, which is currently just for testing rpcs in rpc/blockchain.py. However, I can't think of any better place for it right now and it's definitely an improvement to have this test!

**Comment by achow101** [comment]
ACK 5555fa8b74bc3c8e68232acbf4c54b9a8fadb8f2

File: test/functional/blockchain.py [hunk 2]
```diff
@@ -129,5 +134,18 @@ def _test_getnetworkhashps(self):
         # This should be 2 hashes every 10 minutes or 1/300
         assert abs(hashes_per_second * 300 - 1) < 0.0001
 
+    def _test_stopatheight(self):
+        assert_equal(self.nodes[0].getblockcount(), 200)
+        self.nodes[0].generate(6)
+        assert_equal(self.nodes[0].getblockcount(), 206)
+        self.log.debug('Node should not stop at this height')
+        assert_raises(subprocess.TimeoutExpired, lambda: bitcoind_processes[0].wait(timeout=3))
+        self.nodes[0].generate(1)
```
Comment (sdaftuar) [inline test/functional/blockchain.py]: I believe this call to generate might fail -- isn't it possible the node's shutdown will interrupt the RPC response?  That looks to me like what happened here:
https://travis-ci.org/bitcoin/bitcoin/jobs/245917226#L2275

I think we should wrap the generate(1) in a try...except and allow for the RPC to fail due to connection refused, etc.

**Pull request merged**